### ISSUE TITLE
`useCookie` state is stored independently

### ISSUE THREAD

**Issue by danielroe** [nuxt/nuxt#13020]
### Environment

- Operating System: `Darwin`
- Node Version:     `v14.17.3`
- Nuxt Version:     `3.0.0`
- Package Manager:  `yarn@3.1.1`
- Bundler:          `Vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

```vue
<template>
  <div>
    <h1>
      {{ cookie2 }}
    </h1>
    <button @click="cookie1 = (Math.random() * 100).toFixed()">
      Change cookie
    </button>
  </div>
</template>

<script setup>
const cookie1 = useCookie('test')
const cookie2 = useCookie('test')
</script>
```

### Describe the bug

Multiple invocations of `useCookie` are not synced, and so setting one will not trigger a change to others. This is not a bug per-se, but it's inconsistent compared to `useState`, for example...

**Comment by iamandrewluca** [comment]
It seems this workaround does not work anymore 🤔 

✅ Nuxt v3.3.3
❌ Nuxt v3.4.0

```ts
import { WatchCallback, watch } from 'vue';
import { CookieOptions, CookieRef } from '#app';
import { useCookie, useState } from '#imports';

/**
 * TODO: Remove when is fixed
 * https://github.com/nuxt/nuxt/issues/13020#issuecomment-1397282717
 */
export function useStatefulCookie<T = string | null>(
	name: string,
	options?: CookieOptions<T>,
): CookieRef<T> {
	const cookie = useCookie(name, options);
	const state = useState(name, () => cookie.value);

	const callback: WatchCallback<T, T> = () => {
		cookie.value = state.value;
	};

	watch(state, callback, { deep: true });

	return state;
}
```

**Comment by iamandrewluca** [comment]
Issue above fixed/works in v3.4.1

**Comment by cawa-93** [comment]
May `CookieStore` somehow improve this? `CookieStore` allows you to track when value cookie changed.  Theoretically, `useCookie` may return a ref and add cookie `change` event listener and change ref value. 
```js
function useCookie(name) {
	const cookie = ref()

	// load current cookie value
	const loadCookie = () => ref.value = getCookieCurrentValue(name);
	loadCookie()

	// cookieStore works only in client
	if (process.client) {
		// listen cookie changes and update ref
		cookieStore.addEventListener('change', loadCookie)
		onUnmounted(cookieStore.removeEventListener(loadCookie))
	}

	return cookie;
}
```
